packages/ember-data/lib/system/relationships/ext.js

```diff
@@ -277,9 +278,9 @@ Model.reopenClass({
       var possibleRelationships = relationshipsSoFar || [];
 
       var relationshipMap = get(inverseType, 'relationships');
-      if (!relationshipMap) { return; }
+      if (!relationshipMap) { return possibleRelationships; }
 
-      var relationships = relationshipMap.get(type);
+      var relationships = relationshipMap.get(type.typeKey);
```

Comment:
we should try and make `type` always the string and if its something else call it the factory/modelConstructor or something